--- /dev/null
+++ ../src/FcmMessage.php
@@ -258,7 +258,7 @@
 
     public function toArray()
     {
-        return [
+        $data = [
             'name' => $this->getName(),
             'data' => $this->getData(),
             'notification' => ! is_null($this->getNotification()) ? $this->getNotification()->toArray() : null,
@@ -266,10 +266,23 @@
             'webpush' => ! is_null($this->getWebpush()) ? $this->getWebpush()->toArray() : null,
             'apns' => ! is_null($this->getApns()) ? $this->getApns()->toArray() : null,
             'fcm_options' => ! is_null($this->getFcmOptions()) ? $this->getFcmOptions()->toArray() : null,
-            'token' => $this->getToken(),
-            'topic' => $this->getTopic(),
-            'condition' => $this->getCondition(),
+            // 'token' => $this->getToken(),
+            // 'topic' => $this->getTopic(),
+            // 'condition' => $this->getCondition(),
         ];
+        if ($token = $this->getToken()) {
+            $data['token'] = $token;
+        }
+
+        if ($topic = $this->getTopic()) {
+            $data['topic'] = $topic;
+        }
+
+        if ($condition = $this->getCondition()) {
+            $data['condition'] = $condition;
+        }
+
+        return $data;
     }
 
     /**
